examples/experimental/dagster-airlift/dagster_airlift/core/sensor.py

```diff
@@ -190,3 +188,94 @@ def toposorted_asset_keys(
+        offset=offset,
+    )
+    for i, dag_run in enumerate(runs):
+        peered_dag_asset_info = unmigrated_info.asset_info_by_dag_id[dag_run.dag_id]
+        materializations_for_run = []
+        asset_check_keys_to_request_for_run = set()
+        metadata = {
+            "Airflow Run ID": dag_run.run_id,
+            "Run Metadata (raw)": JsonMetadataValue(dag_run.metadata),
+            "Run Type": dag_run.run_type,
+            "Airflow Config": JsonMetadataValue(dag_run.config),
+            "Creation Timestamp": TimestampMetadataValue(get_current_timestamp()),
+        }
+        # Add dag materialization
+        dag_metadata = {
+            **metadata,
+            "Run Details": MarkdownMetadataValue(f"[View Run]({dag_run.url})"),
+            "Start Date": TimestampMetadataValue(dag_run.start_date),
+            "End Date": TimestampMetadataValue(dag_run.end_date),
+        }
+        materializations_for_run.append(
+            (
+                dag_run.end_date,
+                AssetMaterialization(
+                    asset_key=unmigrated_info.asset_info_by_dag_id[dag_run.dag_id].dag_asset_key,
+                    description=dag_run.note,
+                    metadata=dag_metadata,
+                ),
+            )
+        )
+        for task_run in airflow_instance.get_task_instance_batch(
+            run_id=dag_run.run_id,
+            dag_id=dag_run.dag_id,
+            task_ids=peered_dag_asset_info.task_ids,
+            states=["success"],
+        ):
+            asset_keys = peered_dag_asset_info.asset_keys_for_task(task_run.task_id)
+            task_metadata = {
+                **metadata,
+                "Run Details": MarkdownMetadataValue(f"[View Run]({task_run.details_url})"),
+                "Task Logs": MarkdownMetadataValue(f"[View Logs]({task_run.log_url})"),
+                "Start Date": TimestampMetadataValue(task_run.start_date),
+                "End Date": TimestampMetadataValue(task_run.end_date),
+            }
+            for asset_key in asset_keys:
+                materializations_for_run.append(
+                    (
+                        task_run.end_date,
+                        AssetMaterialization(
+                            asset_key=asset_key,
+                            description=task_run.note,
+                            metadata=task_metadata,
+                        ),
+                    )
+                )
+                asset_check_keys_to_request_for_run.update(
+                    unmigrated_info.checks_per_key[asset_key]
+                )
+
+        asset_check_keys_to_request_for_run.update(
+            unmigrated_info.checks_per_key[peered_dag_asset_info.dag_asset_key]
+        )
+        yield (
+            BatchResult(
+                idx=i + offset,
+                materializations_and_timestamps=materializations_for_run,
+                check_keys=asset_check_keys_to_request_for_run,
```

Comment:
Can the check keys be compute in a separate function that consumes the batch results? 

I'm thinking about how we can more surgically extract this once we can turn DA on for the asset checks.

Reply:
I extracted it into the main loop.